### ISSUE TITLE
GDTCORUploadPackage Memory Leaks 

### ISSUE THREAD

**Issue by ducnnguyen** [firebase/firebase-ios-sdk#5399]
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 11.3.1
  * Firebase SDK version: 6.22.0
  * Firebase Component: Analytics, RemoteConfig, Messaging, Performance
  * Component version: Podfile below
  * Installation method: `CocoaPods `
```
    pod 'Firebase',                 '6.22.0' 
    pod 'Firebase/Analytics'
    pod 'Firebase/RemoteConfig'
    pod 'Firebase/Performance'
    pod 'Firebase/Messaging'
    pod 'nanopb',                   '0.3.9011'
    pod 'GoogleDataTransportCCTSupport', '~> 2.0.3'
```
#### Steps to reproduce:

Our app have homepage with many products, so we call many Firebase functions to tracking events (like impression events, etc..)

We found some memory leaks related to GDTCORUploadPackage
Please help us to resolve this bug.
#### Relevant Code:

![Screen Shot 2020-04-17 at 11 57 17 AM](https://user-images.githubusercontent.com/17038240/79533772-ec8de180-80a2-11ea-8ddf-3542a75fb914.png)
![Screen Shot 2020-04-17 at 11 57 27 AM](https://user-images.githubusercontent.com/17038240/79533775-ee57a500-80a2-11ea-830a-a2a3fd598a5b.png)

**Comment by Sunrise17** [comment]
Yes, i have same issue in 6.22.0 as follows (Xcode version: 11.4.1) ; there was no leakage in previous version.

<img width="1417" alt="GDTCORUploadPackage" src="https://user-images.githubusercontent.com/26038249/79580453-ae5be680-80d1-11ea-8386-010aa85ac9e4.png">

**Comment by morganchen12** [comment]
This is fixed in https://github.com/firebase/firebase-ios-sdk/pull/5382 and will be out in the next release.

**Comment by phuc-nguyen4-tiki** [comment]
@morganchen12 is it fixed on 2.0.3 version? I have updated but nothing improve.

**Comment by mikehaney24** [comment]
No it wasn't, it'll be fixed in GoogleDataTransport v6.0.0

**Comment by dataexcess** [comment]
Hey Hey,

I just came across this issue and wanted to let you know. This fix was not included in 6.24 so I had to hardcode my cocoapod version to '6.23' !

Maybe something to keep in mind.

**Comment by mikehaney24** [comment]
I added unit tests to confirm this particular memory leak was addressed. If you're seeing a memory leak now, please open a new issue and I'll address that.
